warning in sequence_padding.cc

```
[18:20:18]	[ 22%] Building CXX object paddle/operators/math/CMakeFiles/sequence_padding.dir/sequence_padding.cc.o
[18:20:20]	In file included from /paddle/paddle/framework/ddim.h:21:0,
[18:20:20]	from /paddle/paddle/framework/lod_tensor.h:25,
[18:20:20]	from /paddle/paddle/operators/math/sequence_padding.h:17,
[18:20:20]	from /paddle/paddle/operators/math/sequence_padding.cc:15:
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc: In instantiation of 'void paddle::operators::math::PaddingLoDTensorFunctor<paddle::platform::CPUDeviceContext, T>::operator()(const paddle::platform::CPUDeviceContext&, const paddle::framework::LoDTensor&, paddle::framework::Tensor&, bool) [with T = float]':
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:139:16: required from here
[18:20:20]	/paddle/paddle/platform/enforce.h:227:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:108:64: note: in definition of macro 'UNLIKELY'
[18:20:20]	#define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:35:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
[18:20:20]	PADDLE_ENFORCE_EQ(seq_dims[0], abs_offset_lod[level].back(),
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:108:64: note: in definition of macro 'UNLIKELY'
[18:20:20]	#define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:45:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
[18:20:20]	PADDLE_ENFORCE_EQ(padding_dims[0], max_sequence_length,
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:108:64: note: in definition of macro 'UNLIKELY'
[18:20:20]	#define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:50:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
[18:20:20]	PADDLE_ENFORCE_EQ(padding_dims[1], num_sequences,
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:108:64: note: in definition of macro 'UNLIKELY'
[18:20:20]	#define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:55:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
[18:20:20]	PADDLE_ENFORCE_EQ(padding_dims[2], sequence_width,
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc: In instantiation of 'void paddle::operators::math::UnpaddingLoDTensorFunctor<paddle::platform::CPUDeviceContext, T>::operator()(const paddle::platform::CPUDeviceContext&, paddle::framework::LoDTensor&, const paddle::framework::Tensor&, bool) [with T = float]':
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:140:16: required from here
[18:20:20]	/paddle/paddle/platform/enforce.h:227:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:108:64: note: in definition of macro 'UNLIKELY'
[18:20:20]	#define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:96:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
[18:20:20]	PADDLE_ENFORCE_EQ(seq_dims[0], abs_offset_lod[level].back(),
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:108:64: note: in definition of macro 'UNLIKELY'
[18:20:20]	#define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:106:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
[18:20:20]	PADDLE_ENFORCE_EQ(padding_dims[0], max_sequence_length,
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:108:64: note: in definition of macro 'UNLIKELY'
[18:20:20]	#define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:111:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
[18:20:20]	PADDLE_ENFORCE_EQ(padding_dims[1], num_sequences,
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:108:64: note: in definition of macro 'UNLIKELY'
[18:20:20]	#define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
[18:20:20]	^
[18:20:20]	/paddle/paddle/platform/enforce.h:227:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
[18:20:20]	__PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
[18:20:20]	^
[18:20:20]	/paddle/paddle/operators/math/sequence_padding.cc:116:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
[18:20:20]	PADDLE_ENFORCE_EQ(padding_dims[2], sequence_width,
```

--- a/paddle/operators/math/sequence_padding.cc
+++ b/paddle/operators/math/sequence_padding.cc
@@ -32,7 +32,8 @@ class PaddingLoDTensorFunctor<platform::CPUDeviceContext, T> {
     framework::LoD abs_offset_lod = framework::ToAbsOffset(lod);
 
     auto seq_dims = seq.dims();
-    PADDLE_ENFORCE_EQ(seq_dims[0], abs_offset_lod[level].back(),
+    PADDLE_ENFORCE_EQ(seq_dims[0],
+                      static_cast<int64_t>(abs_offset_lod[level].back()),
                       "The first dimension of LoDTensor seq should be "
                       "equal to the sum of all sequences's length.");
 
@@ -41,32 +42,32 @@ class PaddingLoDTensorFunctor<platform::CPUDeviceContext, T> {
                       "The input padding should be a 3-D Tensor of shape "
                       "[max_sequence_length, num_sequences, sequence_width].");
 
-    const size_t max_sequence_length = MaximumSequenceLength(lod, level);
+    const int64_t max_sequence_length = MaximumSequenceLength(lod, level);
     PADDLE_ENFORCE_EQ(padding_dims[0], max_sequence_length,
                       "The first dimension of Tensor padding should be the "
                       "maximum length of all sequences in LoDTensor seq.");
 
-    const size_t num_sequences = abs_offset_lod[level].size() - 1;
+    const int64_t num_sequences = abs_offset_lod[level].size() - 1;
     PADDLE_ENFORCE_EQ(padding_dims[1], num_sequences,
                       "The second dimension of Tensor padding should be the "
                       "number of sequences in LoDTensor seq.");
 
-    const size_t sequence_width = seq.numel() / seq_dims[0];
+    const int64_t sequence_width = seq.numel() / seq_dims[0];
     PADDLE_ENFORCE_EQ(padding_dims[2], sequence_width,
                       "The third dimension of Tensor padding should be the "
                       "width of sequence in LoDTensor seq.");
 
     const T* seq_data = seq.data<T>();
     T* padding_data = padding.data<T>();
-    for (size_t i = 0; i < max_sequence_length; ++i) {
-      for (size_t j = 0; j < num_sequences; ++j) {
-        size_t start_pos = abs_offset_lod[level][j];
-        size_t sequence_length = abs_offset_lod[level][j + 1] - start_pos;
+    for (int64_t i = 0; i < max_sequence_length; ++i) {
+      for (int64_t j = 0; j < num_sequences; ++j) {
+        int64_t start_pos = abs_offset_lod[level][j];
+        int64_t sequence_length = abs_offset_lod[level][j + 1] - start_pos;
         if (i < sequence_length) {
           // i > 0 => sequence_length > 0
           T scale =
               norm_by_times ? (1.0f / static_cast<T>(sequence_length)) : 1.0f;
-          for (size_t k = 0; k < sequence_width; ++k) {
+          for (int64_t k = 0; k < sequence_width; ++k) {
             padding_data[(i * num_sequences + j) * sequence_width + k] =
                 seq_data[(start_pos + i) * sequence_width + k] * scale;
           }
@@ -93,7 +94,8 @@ class UnpaddingLoDTensorFunctor<platform::CPUDeviceContext, T> {
     framework::LoD abs_offset_lod = framework::ToAbsOffset(lod);
 
     auto seq_dims = seq.dims();
-    PADDLE_ENFORCE_EQ(seq_dims[0], abs_offset_lod[level].back(),
+    PADDLE_ENFORCE_EQ(seq_dims[0],
+                      static_cast<int64_t>(abs_offset_lod[level].back()),
                       "The first dimension of LoDTensor seq should be "
                       "equal to the sum of all sequences's length.");
 
@@ -102,31 +104,31 @@ class UnpaddingLoDTensorFunctor<platform::CPUDeviceContext, T> {
                       "The input padding should be a 3-D Tensor of shape "
                       "[max_sequnece_length, num_sequences, sequence_width].");
 
-    const size_t max_sequence_length = MaximumSequenceLength(lod, level);
+    const int64_t max_sequence_length = MaximumSequenceLength(lod, level);
     PADDLE_ENFORCE_EQ(padding_dims[0], max_sequence_length,
                       "The first dimension of Tensor padding should be "
                       "the maximum length of all sequences in LoDTensor seq.");
 
-    const size_t num_sequences = abs_offset_lod[level].size() - 1;
+    const int64_t num_sequences = abs_offset_lod[level].size() - 1;
     PADDLE_ENFORCE_EQ(padding_dims[1], num_sequences,
                       "The second dimension of Tensor padding should be "
                       "the number of sequences in LoDTensor seq.");
 
-    const size_t sequence_width = seq.numel() / seq_dims[0];
+    const int64_t sequence_width = seq.numel() / seq_dims[0];
     PADDLE_ENFORCE_EQ(padding_dims[2], sequence_width,
                       "The third dimension of Tensor padding should be the "
                       "width of sequence in LoDTensor seq.");
 
     const T* padding_data = padding.data<T>();
     T* seq_data = seq.data<T>();
-    for (size_t i = 0; i < num_sequences; ++i) {
-      size_t start_pos = abs_offset_lod[level][i];
-      size_t sequence_length = abs_offset_lod[level][i + 1] - start_pos;
-      for (size_t j = 0; j < sequence_length; ++j) {
+    for (int64_t i = 0; i < num_sequences; ++i) {
+      int64_t start_pos = abs_offset_lod[level][i];
+      int64_t sequence_length = abs_offset_lod[level][i + 1] - start_pos;
+      for (int64_t j = 0; j < sequence_length; ++j) {
         // sequence_width > j > 0
         T scale =
             norm_by_times ? (1.0f / static_cast<T>(sequence_length)) : 1.0f;
-        for (size_t k = 0; k < sequence_width; ++k) {
+        for (int64_t k = 0; k < sequence_width; ++k) {
           seq_data[(start_pos + j) * sequence_width + k] =
               padding_data[(j * num_sequences + i) * sequence_width + k] *
               scale;

--- a/paddle/operators/math/sequence_padding.cu
+++ b/paddle/operators/math/sequence_padding.cu
@@ -71,7 +71,8 @@ class PaddingLoDTensorFunctor<platform::CUDADeviceContext, T> {
     framework::LoD abs_offset_lod = framework::ToAbsOffset(lod);
 
     auto seq_dims = seq.dims();
-    PADDLE_ENFORCE_EQ(seq_dims[0], abs_offset_lod[level].back(),
+    PADDLE_ENFORCE_EQ(seq_dims[0],
+                      static_cast<int64_t>(abs_offset_lod[level].back()),
                       "The first dimension of LoDTensor seq should be "
                       "equal to the sum of all sequences's length.");
 
@@ -80,17 +81,17 @@ class PaddingLoDTensorFunctor<platform::CUDADeviceContext, T> {
                       "The input padding should be a 3-D Tensor of shape "
                       "[max_sequence_length, num_sequences, sequence_width].");
 
-    size_t max_sequence_length = MaximumSequenceLength(lod, level);
+    int64_t max_sequence_length = MaximumSequenceLength(lod, level);
     PADDLE_ENFORCE_EQ(padding_dims[0], max_sequence_length,
                       "The first dimension of Tensor padding should be the "
                       "maximum length of all sequences in LoDTensor seq.");
 
-    const size_t num_sequences = abs_offset_lod[level].size() - 1;
+    const int64_t num_sequences = abs_offset_lod[level].size() - 1;
     PADDLE_ENFORCE_EQ(padding_dims[1], num_sequences,
                       "The second dimension of Tensor padding should be the "
                       "number of sequences in LoDTensor seq.");
 
-    const size_t sequence_width = seq.numel() / seq_dims[0];
+    const int64_t sequence_width = seq.numel() / seq_dims[0];
     PADDLE_ENFORCE_EQ(padding_dims[2], sequence_width,
                       "The third dimension of Tensor padding should be the "
                       "width of sequence in LoDTensor seq.");
@@ -101,7 +102,7 @@ class PaddingLoDTensorFunctor<platform::CUDADeviceContext, T> {
       return;
     }
 
-    const size_t kBlockSize = 512;
+    const int64_t kBlockSize = 512;
 
     /* At least use 32 threads to copy sequence_width elements,
      * and at least 8 elements for each thread.
@@ -143,7 +144,8 @@ class UnpaddingLoDTensorFunctor<platform::CUDADeviceContext, T> {
     framework::LoD abs_offset_lod = framework::ToAbsOffset(lod);
 
     auto seq_dims = seq.dims();
-    PADDLE_ENFORCE_EQ(seq_dims[0], abs_offset_lod[level].back(),
+    PADDLE_ENFORCE_EQ(seq_dims[0],
+                      static_cast<int64_t>(abs_offset_lod[level].back()),
                       "The first dimension of LoDTensor seq should be "
                       "equal to the sum of all sequences's length.");
 
@@ -152,17 +154,17 @@ class UnpaddingLoDTensorFunctor<platform::CUDADeviceContext, T> {
                       "The input padding should be a 3-D Tensor of shape "
                       "[max_sequnece_length, num_sequences, sequence_width].");
 
-    size_t max_sequence_length = MaximumSequenceLength(lod, level);
+    int64_t max_sequence_length = MaximumSequenceLength(lod, level);
     PADDLE_ENFORCE_EQ(padding_dims[0], max_sequence_length,
                       "The first dimension of Tensor padding should be "
                       "the maximum length of all sequences in LoDTensor seq.");
 
-    const size_t num_sequences = abs_offset_lod[level].size() - 1;
+    const int64_t num_sequences = abs_offset_lod[level].size() - 1;
     PADDLE_ENFORCE_EQ(padding_dims[1], num_sequences,
                       "The second dimension of Tensor padding should be "
                       "the number of sequences in LoDTensor seq.");
 
-    const size_t sequence_width = seq.numel() / seq_dims[0];
+    const int64_t sequence_width = seq.numel() / seq_dims[0];
     PADDLE_ENFORCE_EQ(padding_dims[2], sequence_width,
                       "The third dimension of Tensor padding should be the "
                       "width of sequence in LoDTensor seq.");
@@ -173,7 +175,7 @@ class UnpaddingLoDTensorFunctor<platform::CUDADeviceContext, T> {
       return;
     }
 
-    const size_t kBlockSize = 512;
+    const int64_t kBlockSize = 512;
 
     /* At least use 32 threads to copy sequence_width elements,
      * and at least 8 elements for each thread.

